### DISCUSSION TITLE
Modify how suites are counted

### DISCUSSION THREAD

**Discussion by pedrHenrique** [Shelex/cypress-allure-plugin#145] [Q&A] [answered]
Hi,

I would like to know if there is any way that I could change how suites number are counted in the Allure Report generated, in a way that the example below would result In a different number of suites

**Example:**
```
describe(`Specific Test Group`, () => {
    describe(`Test 1`, () => {
        its...
    });
    describe(`Group Test 2 `, () => {
        describe(`Test 2.1`, () => {
            its...
        });
        describe(`Test 2.2`, () => {
            its...
        });
    });
});
```
**Output:()** 
![image](https://user-images.githubusercontent.com/60978132/179096714-a3eff873-dbc6-4020-a049-03c9c025fba8.png)

Instead of returning only 1 suite in the report, I would like to return 3, is it possible?
Counting only the suites or (if present) subSuites.

I'm trying to do this to organize better my report, and this would help a lot me 😁 

Thanks in advance

**Comment by Shelex** [comment, marked as answer]
Hi @pedrHenrique 
Yes, there was long-living feature request for this - #100 , and it seems [Suite structuring](https://github.com/Shelex/cypress-allure-plugin#suite-structuring) feature in v2.29.0 could be used to address your case too.

The function with your custom logic of defining what should be a suite could be placed in `cypress/support/index.js` (or `e2e.js` if you have cypress v10+, in the same file where you do `import '@shelex/cypress-allure-plugin';`).

In order to leave just last suite being displayed in allure you can remove all its' parents with:
```js
Cypress.Allure.reporter
    .getInterface()
    .defineSuiteLabels((titlePath, fileInfo) => {
        return [titlePath.pop()];
    });
```
So it will leave just last describe block: `Test 1`, `Test 2.1`, `Test 2.2`, etc.
